x/evm/types/msg.go

```diff
@@ -198,11 +198,18 @@ func (msg MsgEthereumTx) ValidateBasic() error {
 		return errorsmod.Wrap(err, "failed to unpack tx data")
 	}
 
+	gas := txData.GetGas()
+
 	// prevent txs with 0 gas to fill up the mempool
-	if txData.GetGas() == 0 {
+	if gas == 0 {
 		return errorsmod.Wrap(evmtypes.ErrInvalidGasLimit, "gas limit must not be zero")
 	}
 
+	// prevent gas limit from overflow
+	if g := sdk.NewIntFromUint64(msg.GetGas()); !g.IsInt64() {
```

Comment:
Would add a basic test to sanity check this condition and `gas == 0` if they're not tested already